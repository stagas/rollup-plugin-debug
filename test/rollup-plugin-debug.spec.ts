import path from 'path'
import * as rollup from 'rollup'
import debug from '../src/rollup-plugin-debug'

const fixtures = path.join(__dirname, 'fixtures')

describe('debug', () => {
  it('works', async () => {
    process.env.ROLLUP_DEBUG = 'rollup-plugin-debug:*'
    const bundle = await rollup.rollup({
      input: path.join(fixtures, 'sink.js'),
      plugins: [debug()],
    })
    const result = await bundle.generate({})
    expect(result.output[0].code).toMatchSnapshot()
  })

  it('printId', async () => {
    process.env.ROLLUP_DEBUG = 'rollup-plugin-debug:*'
    const bundle = await rollup.rollup({
      input: path.join(fixtures, 'sink.js'),
      plugins: [debug({ printId: true })],
    })
    const result = await bundle.generate({})
    expect(result.output[0].code).toMatchSnapshot()
  })

  it('ignores', async () => {
    process.env.ROLLUP_DEBUG = ''
    const bundle = await rollup.rollup({
      input: path.join(fixtures, 'sink.js'),
      plugins: [debug()],
    })
    const result = await bundle.generate({})
    expect(result.output[0].code).toMatchSnapshot()
  })

  it('removes', async () => {
    process.env.ROLLUP_DEBUG = ''
    const bundle = await rollup.rollup({
      input: path.join(fixtures, 'sink.js'),
      plugins: [debug({
        removeIfUnmatched: true,
      })],
    })
    const result = await bundle.generate({})
    expect(result.output[0].code).toMatchSnapshot()
  })
})
