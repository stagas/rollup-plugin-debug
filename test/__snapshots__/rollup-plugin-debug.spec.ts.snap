// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debug ignores 1`] = `
"//!? 'hello world'
const someVar = 123;
//!warn 'warning', someVar
//!time 'time'
//!timeEnd 'time'
console.log('sink', someVar);
//!> alert('hello')
"
`;

exports[`debug printId 1`] = `
"console.debug('\\\\x1B[38;2;255;238;136mrollup-plugin-debug:test:fixtures:sink\\\\x1B[m ' + 'hello world');
const someVar = 123;
console.warn('\\\\x1B[38;2;255;238;136mrollup-plugin-debug:test:fixtures:sink\\\\x1B[m ' + 'warning', someVar);
console.time('\\\\x1B[38;2;255;238;136mrollup-plugin-debug:test:fixtures:sink\\\\x1B[m ' + 'time');
console.timeEnd('\\\\x1B[38;2;255;238;136mrollup-plugin-debug:test:fixtures:sink\\\\x1B[m ' + 'time');
console.log('sink', someVar);
alert('hello');
"
`;

exports[`debug removes 1`] = `
"const someVar = 123;



console.log('sink', someVar);
"
`;

exports[`debug works 1`] = `
"console.debug('hello world');
const someVar = 123;
console.warn('warning', someVar);
console.time('time');
console.timeEnd('time');
console.log('sink', someVar);
alert('hello');
"
`;
